### PR TITLE
tools/codespell: fix new typos found by recent version + add new ignored words

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

When a new riotdocker image is built, the latest version of codespell is installed: https://github.com/RIOT-OS/riotdocker/blob/201cb542c1365956b941fa4870ccbd6479392f3d/requirements.txt#L14

And it seems that the latest riotdocker image contains a really new version of codespell with new words added in his dictionary of typos. So when run on master, the static tests are failing because:
- of new false positives (filp and ether)
- new typos

This PR is fixing all of them (luckily, not so much).

### Testing procedure

- Ensure you have the latest release of codespell installed:

```
$ pip install codespell -U
$ codespell --version
2.0.0
```

- Run the codespell check from RIOTBASE: `/dist/tools/codespell/check.sh`. With this PR, you get an empty output, on master a lost of typos are found.

### Issues/PRs references

Last check that fails with the static tests performed on master. ([example job](https://github.com/RIOT-OS/RIOT/runs/1664753077)).

### CODE REVIEW

File: dist/tools/codespell/ignored_words.txt [hunk 1]
```diff
@@ -106,7 +106,10 @@ dedup
...
 # Chang (common name) => Change
-Chang
-
-# chang (common name) => change
 chang
+
+# filp (variable name for file pointer) => flip
+filp
+
+# Ether (Scapy submodule name and means Ethernet in some parts of the code) => Either
```
Comment (miri64) [inline dist/tools/codespell/ignored_words.txt:114]: [It's a class](https://github.com/secdev/scapy/blob/master/scapy/layers/l2.py#L211), not a submodule ;-).

**Review by miri64 — CHANGES_REQUESTED** [review]
Minor nit

File: dist/tools/codespell/ignored_words.txt (hunk 1 above)
Comment (aabadie) [reply dist/tools/codespell/ignored_words.txt]: Should be fixed (directly squashed)

**Review by miri64 — APPROVED** [review]
ACK

File: cpu/esp32/include/periph_cpu.h [hunk 2]
```diff
@@ -498,7 +498,7 @@ typedef struct {
 
 
 /**
- * @name   Timer configuration depenend on which implementation is used
+ * @name   Timer configuration depend on which implementation is used
```
Comment (gschorcht) [inline cpu/esp32/include/periph_cpu.h:501]: The intention was to say "Timer configuration depending on which implementation used"

Comment (miri64) [reply cpu/esp32/include/periph_cpu.h:501]: Can you please fix that in a follow-up?

Comment (gschorcht) [reply cpu/esp32/include/periph_cpu.h:501]: Sure, but I have no idea yet when I will have time for that. The review process was too fast to react in time. :wink:

Comment (aabadie) [reply cpu/esp32/include/periph_cpu.h:501]: Sorry about that. Just ACK #15727 and it will be good :)

**Pull request merged**